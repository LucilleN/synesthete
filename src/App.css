.App {
  /* margin-left: 5rem;
  margin-right: 5rem; */
  width: 100%;
  /* padding-left: 5rem;
  padding-right: 5rem; */
}
