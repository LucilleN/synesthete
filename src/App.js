import React, { useEffect } from 'react'
import { Route, BrowserRouter as Router, Switch } from 'react-router-dom'

import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles'
import { withStyles } from '@material-ui/core/styles'

import Homepage from './pages/Homepage'
import About from './pages/About'
import Search from './pages/Search'
import UploadVisualization from './pages/UploadVisualization'
import Visualization from './pages/Visualization'
import MenuBar from './components/MenuBar'

import { apiHost } from './api'

export const theme = createMuiTheme({
  palette: {
    primary: {
      main: '#c23063', // magenta
      dark: '#180142' // dark purple
    },
    type: 'dark',
    dark: {
      pink: '#bf265c',
      purple: '#1c004f',
    },
    medium: {
      purple: '#5a0b66',
    },
    light: {
      pink: '#db91aa',
      purple: '#ac91db'
    },
    white: '#FFFFFF',
  },
  typography: {
    fontFamily: ['Dosis', 'Roboto', 'Open Sans', '"Helvetica Neue"', 'Arial', 'sans-serif'].join(',')
  }
})

export const defaultErrorText = 'Sorry, but something went wrong.'

const styles = () => ({
  root: {
    width: '100vw',
    height: '100%',
    minHeight: '100vh',
    background: `linear-gradient(0deg, ${theme.palette.dark.pink} 0%, ${theme.palette.dark.purple} 80%)`
  }
})

const routes = (
  <Router>
    <div>
      <MenuBar />
      <Switch>
        <Route exact path="/" component={Homepage} />
        <Route exact path="/about" component={About} />
        <Route exact path="/search" component={Search} />
        <Route exact path="/upload" component={UploadVisualization} />
        <Route exact path="/visualization" component={Visualization} />
        <Route exact path="/visualization/:songID" component={Visualization} />
      </Switch>
    </div>
  </Router>
)

const App = (props) => {

  useEffect(() => apiHost('https://api.spotify.com/v1/'))

  const { classes } = props

  return (
    <div className={classes.root}>
      <MuiThemeProvider theme={theme}>
        {routes}
      </MuiThemeProvider>
    </div>
  )
}

export default withStyles(styles)(App)
