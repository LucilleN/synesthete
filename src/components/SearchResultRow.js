import React, { useState } from 'react'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'

import { Redirect } from 'react-router-dom';

const styles = theme => ({
  row: {
    background: theme.palette.light.pink,
    padding: '4px 16px',
    margin: 2,
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    '&:hover': {
       background: theme.palette.white,
    },
  },
  text: {
    color: theme.palette.dark.purple,
    '&:hover': {
       color: theme.palette.dark.purple
    },
  }
})

const SearchResultRow = props => {

  const { classes, song } = props

  const [clicked, setClicked] = useState(false)

  const handleClick = () => {
    console.log("SEARCH RESULT ROW", "handleClick called")
    setClicked(true)
  }

  if (!clicked) {
    return (
      <Grid item xs={12} id="search-result-row" className={classes.row} onClick={handleClick}>
        <Typography className={classes.text}>
          "{song.name}," Artist: {song.artists[0].name}
        </Typography>
        <Typography>
          {">>> Visualize This Sample >>>"}
        </Typography>
      </Grid>
    )
  }
  else {
    return (
      <Redirect to={{
          pathname: `/visualization/${song.id}`,
          state: { trackObject: song }
        }}
      />
    )
  }

}

export default withStyles(styles)(SearchResultRow)
