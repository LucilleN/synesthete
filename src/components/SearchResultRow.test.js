import React from 'react'
import ReactDOM from 'react-dom'
import TestRenderer from 'react-test-renderer'
import ReactTestUtils from 'react-dom/test-utils'
import { MuiThemeProvider } from '@material-ui/core/styles'
import { Route, BrowserRouter as Router } from 'react-router-dom';

import { theme } from '../App'
import SearchResultRow from './SearchResultRow'

let component;
beforeEach(() => {
  component = TestRenderer.create(
    <MuiThemeProvider theme={theme}>
      <SearchResultRow />
    </MuiThemeProvider>
  )
})

describe('initial state', () => {
  it('should render the song title and artist name', () => {
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('when it is clicked', () => {
  let div
  beforeEach(() => {
    div = document.createElement('div')
    ReactTestUtils.act(() => {
      ReactDOM.render(
        <Router>
          <MuiThemeProvider theme={theme}>
            <SearchResultRow />
          </MuiThemeProvider>
        </Router>
      , div)
    })
    const searchResultRow = div.querySelector('#search-result-row')
    ReactTestUtils.act(() => {
      ReactTestUtils.Simulate.click(searchResultRow)
    })
  })

  afterEach(() => ReactDOM.unmountComponentAtNode(div))

  it('should no longer render the search result row after the redirect', () => {
    const searchResultRow = div.querySelector('#search-result-row')
    expect(searchResultRow).toBeNull()
  })

})
