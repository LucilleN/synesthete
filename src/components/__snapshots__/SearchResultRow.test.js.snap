// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial state should render the song title and artist name 1`] = `
<div
  className="MuiGrid-root SearchResultRow-row-1 MuiGrid-item MuiGrid-grid-xs-12"
  id="search-result-row"
  onClick={[Function]}
>
  <p
    className="MuiTypography-root SearchResultRow-text-2 MuiTypography-body1"
  >
    "
    The Night We Met
    ," Artist: 
    Lord Huron
  </p>
  <p
    className="MuiTypography-root MuiTypography-body1"
  >
    &gt;&gt;&gt; Visualize This Sample &gt;&gt;&gt;
  </p>
</div>
`;
